jetty-websocket/websocket-common/src/main/java/org/eclipse/jetty/websocket/common/message/SimpleBinaryMessage.java

```diff
@@ -23,20 +23,22 @@
 import java.nio.ByteBuffer;
 
 import org.eclipse.jetty.util.BufferUtil;
+import org.eclipse.jetty.util.ByteArrayOutputStream2;
 import org.eclipse.jetty.websocket.common.events.EventDriver;
 
 public class SimpleBinaryMessage implements MessageAppender
 {
+    private static final ByteArrayOutputStream2 EMPTY_BUF = new ByteArrayOutputStream2(0);
```

Comment:
To ensure that `out` is never `null` and accomodate for `o.e.j.w.j.m.BinaryWholeMessage` overriding `messageComplete`, it's initialized with a sentinel value. It's also possible to override `write` to ensure that no other messages can `write` into the sentinel value if this can be a concern (I'm not sure whether extending `SimpleBinaryMessage` can be considered a public API).

Reply:
Still seems a little weird to use a static `EMPTY_BUF` when it is mutable.

I would be more inclined to use `null` and then in `BinaryWholeMessage.messageComplete()` do something like
```java
byte[] data = (out == null) ? new byte[0] : out.toByteArray();
```

This shouldn't be considered public API it is an implementation class.

Reply:
Hey, @lachlan-roberts! Thanks, the comment addressed in https://github.com/eclipse/jetty.project/pull/4564/commits/66f4bda2eaac89d8b6696937de25aa3ce8c50505.